flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequestExecutorImplTest.java

```diff
@@ -0,0 +1,203 @@
+		ChannelStateWriteRequestExecutorImpl worker = new ChannelStateWriteRequestExecutorImpl(NO_OP, closingDeque);
+		closingDeque.setWorker(worker);
+		TestWriteRequest request = new TestWriteRequest();
+		requestFun.accept(worker, request);
+		assertTrue(closingDeque.isEmpty());
+		assertFalse(request.isCancelled());
+	}
+
+	private void testSubmitFailure(BiConsumerWithException<ChannelStateWriteRequestExecutor, ChannelStateWriteRequest, Exception> submitAction) throws Exception {
+		TestWriteRequest request = new TestWriteRequest();
+		LinkedBlockingDeque<ChannelStateWriteRequest> deque = new LinkedBlockingDeque<>();
+		try {
+			submitAction.accept(new ChannelStateWriteRequestExecutorImpl(NO_OP, deque), request);
+		} catch (IllegalStateException e) {
+			// expected: executor not started;
+			return;
+		} finally {
+			assertTrue(request.cancelled);
+			assertTrue(deque.isEmpty());
+		}
+		throw new RuntimeException("expected exception not thrown");
+	}
+
+	@Test
+	@SuppressWarnings("CallToThreadRun")
+	public void testCleanup() throws IOException {
+		TestWriteRequest request = new TestWriteRequest();
+		LinkedBlockingDeque<ChannelStateWriteRequest> deque = new LinkedBlockingDeque<>();
+		deque.add(request);
+		TestRequestDispatcher requestProcessor = new TestRequestDispatcher();
+		ChannelStateWriteRequestExecutorImpl worker = new ChannelStateWriteRequestExecutorImpl(requestProcessor, deque);
+
+		worker.close();
+		worker.run();
+
+		assertTrue(requestProcessor.isStopped());
+		assertTrue(deque.isEmpty());
+		assertTrue(request.isCancelled());
+	}
+
+	@Test
+	public void testIgnoresInterruptsWhileRunning() throws Exception {
+		TestRequestDispatcher requestProcessor = new TestRequestDispatcher();
+		LinkedBlockingDeque<ChannelStateWriteRequest> deque = new LinkedBlockingDeque<>();
+		try (ChannelStateWriteRequestExecutorImpl worker = new ChannelStateWriteRequestExecutorImpl(requestProcessor, deque)) {
+			worker.start();
+			worker.getThread().interrupt();
+			worker.submit(new TestWriteRequest());
+			worker.getThread().interrupt();
+			while (!deque.isEmpty()) {
+				Thread.sleep(100);
+			}
+		}
+	}
+
+	@Test
+	public void testCanBeClosed() throws IOException {
+		TestRequestDispatcher requestProcessor = new TestRequestDispatcher();
+		try (ChannelStateWriteRequestExecutorImpl worker = new ChannelStateWriteRequestExecutorImpl(requestProcessor)) {
+			worker.start();
+		}
+	}
+
+	@Test
+	public void testRecordsException() throws Exception {
+		TestException testException = new TestException();
+		TestRequestDispatcher requestProcessor = new TestRequestDispatcher() {
+			@Override
+			public void dispatch(ChannelStateWriteRequest request) {
+				throw testException;
+			}
+		};
+		LinkedBlockingDeque<ChannelStateWriteRequest> deque = new LinkedBlockingDeque<>(Arrays.asList(new TestWriteRequest()));
+		try (ChannelStateWriteRequestExecutorImpl worker = new ChannelStateWriteRequestExecutorImpl(requestProcessor, deque)) {
```

Comment:
I am not quite clear what is this test motivation for. When the dispatcher throws exception, it would terminate the internal thread inside `ChannelStateWriteRequestExecutorImpl`. And when the `worker#close`, it should throw exception actually, i think we should verify the exception is same with `testException`

Reply:
Right.
Fixed, thanks for pointing this out